src/Features/ERDDAP/Platform/Observations/WindCondition/index.tsx

```diff
@@ -1,90 +1,92 @@
 /**
  * Wind Observed conditions component
  */
-import * as Sentry from "@sentry/react"
 import * as React from "react"
-import { SizeMeProps, withSize } from "react-sizeme"
 import { Alert, Col, Row } from "reactstrap"
 
 import { WindTimeSeriesChart } from "components/Charts"
+import { useUnitSystem } from "Features/Units"
 import { UnitSystem } from "Features/Units/types"
 import { DataTimeSeries } from "Shared/timeSeries"
 
-import { PlatformFeatureWithDatasets } from "../../../types"
-import { ErddapDatasetLoader, ErddapDatasetStatus } from "../../Dataset"
+import { PlatformFeature, PlatformTimeSeries } from "../../../types"
+import { pickWindDatasets, pickWindTimeSeries } from "../../../utils/wind"
+import { useDatasets } from "Features/ERDDAP/hooks"
 
 interface Props {
-  platform: PlatformFeatureWithDatasets
+  platform: PlatformFeature
+}
+
+interface LoadingProps extends Props {
+  timeSeries: PlatformTimeSeries[]
+}
+
+interface DisplayProps extends LoadingProps {
   unit_system: UnitSystem
+  datasets: DataTimeSeries[]
 }
 
-export const windStandards = new Set(["wind_from_direction", "wind_speed", "wind_speed_of_gust"])
+export const ErddapWindObservedCondition: React.FunctionComponent<Props> = ({ platform }) => {
+  const { timeSeries } = pickWindTimeSeries(platform)
 
-export const ErddapWindObservedConditionBase: React.SFC<Props & SizeMeProps> = ({ platform, size, unit_system }) => {
-  // adjust number of barbs based on width
-  let barbsPerDay = 5
-  if (size && size.width && size.width < 800) {
-    barbsPerDay = 2
+  if (timeSeries.length < 0) {
+    return <WindError message="No wind data" />
   }
 
-  const windDatasets = platform.properties.readings.filter((reading) =>
-    windStandards.has(reading.data_type.standard_name)
-  )
+  return <LoadWindObservedConditionDisplay platform={platform} timeSeries={timeSeries} />
```

Comment:
Codacy found an issue: ['LoadWindObservedConditionDisplay' was used before it was defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)